.github/workflows/version_upgrade.yml

```diff
@@ -0,0 +1,53 @@
+# This workflow is used in bump_version and release workflows. It
+# install poetry and updates the version of the code. It gives back
+# the new version which can be used in the consequent workflows.
+
+name: Version upgrade with poetry
+
+on:
+  workflow_call:
+    inputs:
+      release_type:
+        required: true
+        type: string
+    outputs:
+      project_version:
+        value: ${{ jobs.upgrade-version.outputs.job_output_version}}
+
+
+jobs:
+  upgrade-version:
+    runs-on: ubuntu-latest
+    outputs:
+      job_output_version: ${{ steps.get-version.outputs.PROJECT_VERSION}}
+    steps:
+      - name: checkout code
+        uses: actions/checkout@v4
+        with:
+          token: ${{ secrets.VERSION_DECOIMPACT }}
+          fetch-depth: 0
+      - name: install python
+        uses: actions/setup-python@v4
+        with:
+          python-version: 3.11
```

Comment:
is this on purpose hardcoded here?

Reply:
Before: yes. Before ticket DEI-184 (update packages), this was the highest supported python version, due to scipy dependencies.. Hidde upgraded everything and those scipy issues are fixed and we can use latest python version now. 
Now: I'm not sure if we should move away from a hardcoded version (but totally agree on higher version) so we always have a stable build. Otherwise this might break when upgrading packages again